### ISSUE TITLE
Error: Could not find a compiler version matching 0.7.2.

### ISSUE THREAD

**Issue by vittominacori** [trufflesuite/truffle#3404]
## Issue

I tried to use solc 0.7.2 as I see it is available on solc-bin. 
I received `Could not find a compiler version matching 0.7.2` and if I run truffle with `--list` option it says latest is 0.7.1.

## Steps to Reproduce

Just set `solc.version: 0.7.2` in yout truffle-config.js and compile.

## Expected Behavior

Download solc and compile with 0.7.2 as usual.

## Actual Results

```
truffle(develop)> compile
Error: Error: Could not find a compiler version matching 0.7.2. Please ensure you are specifying a valid version, constraint or build in the truffle config. Run `truffle compile --list` to see available versions.
    at VersionRange.load (./node_modules/truffle/build/webpack:/packages/compile-solidity/compilerSupplier/loadingStrategies/VersionRange.js:199:1)
```

## Environment

```
truffle(develop)> version
Truffle v5.1.46 (core: 5.1.46)
Solidity - 0.7.2 (solc-js)
Node v10.15.3
Web3.js v1.2.1
```

**Comment by haltman-at** [comment]
Hm, yup, this is happening to me as well.

**Comment by vittominacori** [comment]
It seems that it depends on compilerRoots default url. 

https://solc-bin.ethereum.org/bin/ should be used instead of https://ethereum.github.io/solc-bin/bin/

**Comment by haltman-at** [comment]
OK, this should be fixed now with today's release.  Closing it.  Thanks for bringing this up!

**Comment by ivan-homoliak-sutd** [comment]
Hi guys, I am still getting this error with truffle v5.1.60 as well as some downgraded versions v5.1.59 and v5.1.58.
The error is the same for any other compiler than the latest 0.7.x, which is 0.7.6.
Are you sure that it is fixed?

**Comment by haltman-at** [comment]
Hm, I believe this was supposed to have been fixed in #3406 (not sure how that never got linked here).

I can't reproduce the problem, but I'll reopen this all the same I guess.  Do you have more detailed reproduction steps, perhaps?

**Comment by ivan-homoliak-sutd** [comment]
Ok, I reinstalled Node through nvm and reinstalled truffle, and it already works. Although, I am not sure why it was fixed. Before, I was using the Node & npm installed through Ubuntu package manager but due to some permission issue when installing truffle, I repeated it with nvm. 

**Steps to reproduce:** I set up compiler version in truffle-config.js and run truffle compile, which attempted to download requested versions of solc - I've seen two attempt - each time waiting maybe 20 seconds and then message _Could not find a compiler version matching 0.7.2_

I was wondering whether there might be some permission issues when downloading the compiler but this would not explain long waiting. What would be great if truffle would have --verbose option for future cases.

For now, we can close.
